### PR TITLE
Fix juice stream droplets spawning in incorrect locations

### LINKED ISSUE #2149: osu!catch generates a TinyDroplet at the stream's start time
This is not intended functionality, as there are essentially two objects at the start time of a juice stream: The leading starting Fruit, and a TinyDroplet.

### PR DESCRIPTION
**Pull request by peppy**
Closes #2149.

### CODE REVIEW

File: osu.Game.Rulesets.Catch/Tests/CatchBeatmapConversionTest.cs [hunk 1]
```diff
@@ -18,7 +17,6 @@ public class CatchBeatmapConversionTest : BeatmapConversionTest<ConvertValue>
     {
         protected override string ResourceAssembly => "osu.Game.Rulesets.Catch";
 
-        [TestCase("basic"), Ignore("See: https://github.com/ppy/osu/issues/2149")]
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Catch/Tests/CatchBeatmapConversionTest.cs]: Not intended?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Catch/Tests/CatchBeatmapConversionTest.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Rulesets.Catch/Tests/CatchBeatmapConversionTest.cs]: Not intended.

**Comment by peppy** [comment]
Restored ignored test for now as there's deeper issues.

**Comment by smoogipoo** [comment]
This seems okay for now. I'm a little bit weary of the minDistanceToEnd calculation, and if that matches stable, but need FastRandom usage to be fixed and additional tests to be done.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**